backend/benefit/messages/views.py

```diff
@@ -0,0 +1,53 @@
+from applications.models import Application
+from common.permissions import BFIsAuthenticated, BFIsHandler, TermsOfServiceAccepted
+from django.conf import settings
+from django.utils.translation import gettext_lazy as _
+from messages.models import Message
+from messages.permissions import IsMessageAuthor
+from messages.serializers import MessageSerializer, NoteSerializer
+from rest_framework import viewsets
+from rest_framework.exceptions import NotFound
+from users.utils import get_company_from_request
+
+
+class ApplicantMessageViewSet(viewsets.ModelViewSet):
+
+    serializer_class = MessageSerializer
+    permission_classes = [BFIsAuthenticated, TermsOfServiceAccepted, IsMessageAuthor]
+
+    def get_queryset(self):
+        if settings.DISABLE_AUTHENTICATION:
+            return Application.objects.get(
+                id=self.kwargs["application_pk"]
+            ).messages.get_messages_qs()
+        company = get_company_from_request(self.request)
+        try:
+            application = company.applications.get(id=self.kwargs["application_pk"])
+        except Application.DoesNotExist:
+            raise NotFound(_("Application not found"))
+        if company == application.company:
```

Comment:
Wouldn't check would always be True (as we are querying company.applications) so code can be simplified?

Reply:
Nice catch, removed this part